p5js/NeuralNetwork/NeuralNetwork_co2net-training/sketch.js

```diff
@@ -10,253 +10,111 @@ This example uses a callback pattern to create the classifier
 // x: population_cdp
 // y: scope1_ghg_emissions_tons_co2e
 === */
-let neuralNetwork;
+let nn;
 let data;
 let predictions = [];
-const margins = 40;
+// const margins = 40;
+
+// const population = [
+//   100, 
+//   10000,
+//   100000,
+//   1000000,
+//   10000000,
+//   20000000,
+//   30000000
+// ]
+
+const population = [
+  0.1, 
+  0.2,
+  0.3,
+  0.4,
+  0.5,
+  0.6,
+  0.7
+]
+
+
 
 // Options for Neural Network
 const options = {
-  inputs: 1,
-  outputs: 1,
-  inputKeys: ['population_cdp'],
-  outputKeys: ['scope1_ghg_emissions_tons_co2e']
-  // activation: 'sig'
+  inputs: ['population_cdp'],
+  outputs: ['scope1_ghg_emissions_tons_co2e'],
+  dataUrl:'data/co2stats.csv',
+  task:'regression',
+  epochs: 300,
+  batchSize:24
 };
 
-const DATA_URL = 'data/co2stats.csv'
 
 function setup() {
   createCanvas(400, 400);
   // background(0, 27, 68);
   background(244, 244, 244);
 
+  console.log('hello!')
   // Step 1: Create Neural Network
-  neuralNetwork = ml5.neuralNetwork(options);
-
-  // Step 2: Prepare the data
-  neuralNetwork.loadData(DATA_URL, dataLoaded);
-  // prepData()
-
-  // Step 3: inspect the data
-  // inspectData(data.normalized_input, data.normalized_target);
-
-  // Step 4A: Inspect the data
-  // trainModel()
-
-  // Step 4B: Load a pre-trained model
-  // visualize the predictions;
-  // the population input needs to be log10 transformed
-  // neuralNetwork.load('model/model.json', function() {
-  //   console.log('model loaded')
-
-  //   for(let i = 500; i < 10000000; i*=2){
-  //     predict(Math.log10(i));  
-  //   }
-  //   // vancouver
-  //   predict(Math.log10(603500));
-  //   // NYC
-  //   predict(Math.log10(8537673));
-  // });
-
-}
-
-function dataLoaded(err, results){
-  data = results
-  normalizedData = neuralNetwork.normalize(data);
-  console.log(normalizedData)
-
-  neuralNetwork.train(normalizedData.inputs, normalizedData.labels, finishedTraining)
-}
-
+  nn = ml5.neuralNetwork(options, modelLoaded);
```

Comment:
This is such a tiny detail and I'm not sure what is technically correct, but I might go with `modelReady()` (or `modelCreated()` since `modelLoaded()` to me implies loading a pre-trained model?

```suggestion
  nn = ml5.neuralNetwork(options, modelReady);
```

Reply:
Yep this is also resolved.